[WIP] Implement onboarding workflow

hebasto opened the pull request:
This PR implements **Bosch-0**'s design:

![Screenshot from 2021-10-02 21-42-07](https://user-images.githubusercontent.com/32963518/135728652-1e6bb078-b199-496a-9443-ec0d5bb868cb.png)

Based on #42, and only the last commit belongs to this PR.

hebasto commented:
cc @Bosch-0

promag commented on src/qml/pages/onboarding/Onboarding00Welcome.qml:
```diff
@@ -0,0 +1,71 @@
+    spacing: 16
+
+    Image {
+        id: appLogo
```
21c969272e718e63d3de00a9fe9e0f702eb4fd9f

Let's avoid unnecessary names (`id:`).

I also think these should be snake_case to distinguish from property names.

hebasto replied:
> Let's avoid unnecessary names (`id:`).

Agree.

> I also think these should be snake_case to distinguish from property names.

I'd prefer to follow QML style provided by Qt docs. It guarantees more consistency in long run with more contributors.

hebasto replied:
> Let's avoid unnecessary names (`id:`).

[Done](https://github.com/bitcoin-core/gui-qml/pull/43#issuecomment-933918510).

promag commented on src/qml/pages/onboarding/Onboarding00Welcome.qml:
```diff
@@ -0,0 +1,71 @@
+        sourceSize.height: 128
+    }
+
+    Text {
```
21c969272e718e63d3de00a9fe9e0f702eb4fd9f

Should use `Label` instead.

hebasto replied:
`Label` extends `Text`. Which `Label`'s properties we are interested in?

promag replied:
I’d say background and palette.

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin-core/gui-qml/pull/43#issuecomment-933918510).

promag commented on src/qml/pages/onboarding/Onboarding00Welcome.qml:
```diff
@@ -0,0 +1,71 @@
+import QtQuick 2.12
+import QtQuick.Controls 2.12
+
+Column {
```
21c969272e718e63d3de00a9fe9e0f702eb4fd9f

We should extend `Pane`, something like:

```qml
Pane {
    contentItem: ColumnLayout {
        // ...
    }
}
```

hebasto replied:
> We should extend `Pane`...

Why? We already have `ApplicationWindow` container.

hebasto replied:
[Updated](https://github.com/bitcoin-core/gui-qml/pull/43#issuecomment-933995134).

promag reviewed (commented):
Pushed my suggestions to https://github.com/promag/gui-qml/tree/qml-43. See 

https://user-images.githubusercontent.com/3534524/135757737-be096e27-96db-40cc-840c-a9f236590ebe.mov

jarolrod commented:
> Pushed my suggestions to https://github.com/promag/gui-qml/tree/qml-43. See
> 
>  Screen.Recording.2021-10-03.at.15.14.27.mov

@Bosch-0 @GBKS In regards to this video: Shouldn't the windows have a minimum size, and minimum amount of padding?

promag commented:
> > Pushed my suggestions to https://github.com/promag/gui-qml/tree/qml-43. See
> > Screen.Recording.2021-10-03.at.15.14.27.mov
> 
> @Bosch-0 @GBKS In regards to this video: Shouldn't the windows have a minimum size, and minimum amount of padding?

Definitely, this is just to show that layouts is superior to anchors.

jarolrod commented:
@promag oh right, sorry!

hebasto commented:
@promag
> Pushed my suggestions to https://github.com/promag/gui-qml/tree/qml-43.

`wrapMode: Text.WordWrap` does not work as expected.

hebasto commented:
Updated 21c969272e718e63d3de00a9fe9e0f702eb4fd9f -> 7fc49d75c6da0a1fe700458e4130ad18df3a04ae ([pr43.01](https://github.com/hebasto/gui-qml/commits/pr43.01) -> [pr43.02](https://github.com/hebasto/gui-qml/commits/pr43.02)):

- rebased
 - addressed some of @promag's comments

hebasto commented:
Updated 7fc49d75c6da0a1fe700458e4130ad18df3a04ae -> 6a19c31561c658c961f593faef5d151c1e68ef1d ([pr43.02](https://github.com/hebasto/gui-qml/commits/pr43.02) -> [pr43.04](https://github.com/hebasto/gui-qml/commits/pr43.04)):

- implemented @promag's suggestions to use `ColumnLayout` and `Pane`

Added commits are supposed to be squashed before merging.

promag commented on src/qml/pages/onboarding/Onboarding00Welcome.qml:
```diff
@@ -0,0 +1,72 @@
+import QtQuick.Layouts 1.12
+
+Pane {
+    background: Rectangle {
```
Not needed? The window background is already black. To improve rendering time make `background: null`

promag replied:
Documentation for null or for rendering improvement?

jarolrod commented on src/qml/pages/onboarding/Onboarding00Welcome.qml:
```diff
@@ -0,0 +1,72 @@
+
+        Image {
+            Layout.alignment: Qt.AlignCenter
+            source: "image://images/app"
```
making a note that @bosch-0 has a proposal for a new icon. It needs to be discussed at greater length with the community; if there is consensus we should use it: https://github.com/bitcoin-core/gui/pull/199

Bosch-0 replied:
It's not just a new icon, there is also some changes to the watermark and thus overall logo. See this file for details: https://www.figma.com/file/c1V7b23n0LqRbVJlUkE1mn/Bitcoin-Core-App-Bosch?node-id=1850%3A44

GBKS replied:
The new proposed icon is visually not centered in the circle. Can we address that, please?

Also, there are two oranges in the Figma file, Bitcoin Core orange and Bitcoin orange. They are slightly different, which one is the "right one"?

Bosch-0 replied:
I updated that awhile ago, looks visually centered to me?

![image](https://user-images.githubusercontent.com/55287964/138221699-019358a0-f595-46b5-949e-7bdf1774bf4b.png)


The Bitcoin Core orange is the primary orange, the Bitcoin orange is just there to illustrate the orange used by the Bitcoin logo - ill remove it from the stylesheet.

jarolrod commented on src/qml/pages/onboarding/Onboarding00Welcome.qml:
```diff
@@ -0,0 +1,72 @@
+
+        Label {
+            Layout.maximumWidth: appTitle.width
+            text: "Be part of the Bitcoin network."
```
wrap in `qsTr` in preparation for translation support?
```suggestion
            text: qsTr("Be part of the Bitcoin network.")
```

hebasto commented:
Closing in favor of #124.
